wandb/sdk/launch/agent/agent.py

```diff
@@ -589,6 +589,21 @@ def _check_run_finished(
         try:
             run = job_tracker.run
             status = run.get_status().state
+
+            # Ensure run was not stopped intentionally
+            if status == "preempted":
+                try:
+                    run_state = RunState(
+                        self._api.get_run_state(
+                            self._entity, job_tracker.project, job_tracker.run_id
+                        )
+                    )
+                    if not run_state.is_alive:
+                        status = "stopped"
+                except wandb.CommError:
```

Comment:
can we log a message here, just in case we actually do get a legit commError?

```suggestion
                except wandb.CommError as e:
                    _logger.info(f"failed to get runstate for run {job_tracker.run_id} : {e}")
```